json-rpc/src/stream_rpc/subscriptions.rs

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! # Subscription Handlers
+//!
+//! ## Implementing the Subscription Trait
+//! In order to turn a subscription parameter struct into something which is capable of launching
+//! a subscription, you must implement the `Subscription<ParamType>` trait.
+//! To do so, there are three methods you must define:
+//! 1. `init(&mut self, helper: &SubscriptionHelper) -> Result<(), JsonRpcError>;`
+//!     - This is where initial state is set up to keep track of what has been sent to the client.
+//!       This could be any incremental id: a sequence number for events, the transaction version for transactions, etc.
+//!     - In order to store these variables, fields can be added to the struct:
+//!         1. Whose type implements the `Default` trait and has a `::Default` method
+//!         2. Are marked with a `#[serde(skip)]`
```

Comment:
This seems not ideal? If I'm understanding this correctly if you have a field you need specifically for your implementation of Subscription then you need to add a field, which isn't part of the request arguments, to the param structs?

Reply:
Separated :-)